File: src/learning_topic/scripts/person_subscriber.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# 该例程将订阅/person_info话题，自定义消息类型learning_topic::Person

import rospy
from learning_topic.msg import Person

def personInfoCallback(msg):
    rospy.loginfo("Publish person message[%s, %d %d]",
            msg.name, msg.age, msg.sex)
    
def person_publisher():
    rospy.init_node('person_subscriber', anonymous=True)

    rospy.Subscriber('/person_info', Person, personInfoCallback)

    rospy.spin()


if __name__ == '__main__':
    person_publisher()

File: src/learning_topic/src/person_subscriber.cpp
/**
 * 该例程将订阅/person_info话题，自定义消息类型learning_topic::Person
 */

#include <ros/ros.h>
#include "learning_topic/Person.h"

void personInfoCallback(const learning_topic::Person::ConstPtr& msg){
    
    ROS_INFO("Subscribe Person Info: name:%s  age:%d  sex:%d",
            msg->name.c_str(), msg->age, msg->sex);
 }
 
int main(int argc, char *argv[])
{
    // ROS节点初始化
    ros::init(argc, argv, "person_subscriber");

    ros::NodeHandle n;

    ros::Subscriber person_info_sub = n.subscribe("/person_info", 10, personInfoCallback);

    ros::spin();

    return 0;
}